### PR TITLE
feat(ui): support notification popover at navbar

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes :

I worked on Notification changes for the new UI 
Closes: #

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
1. Notification within app:
Screenshot 2022-07-20 at 7 50 38 PM

2. Notification from browser:

Screenshot 2022-07-20 at 7 50 33 PM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

**Comment by chirag-madlani** [comment]
DO NOT MERGE AS Backend changes isn't pushed yet

File: openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  const currentUser = useMemo(
+    () => AppState.getCurrentUserDetails(),
+    [AppState.userDetails, AppState.nonSecureUserDetails]
+  );
+  const [notifications, setNotifications] = useState<Thread[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const [viewAllPath, setViewAllPath] = useState<string>(
+    `${getUserPath(currentUser?.name as string)}/tasks?feedFilter=ASSIGNED_TO`
+  );
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx:49]: Can we create an enum for this filter and use it here?

File: openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  ) => {
+    setIsLoading(true);
+    getFeedsWithFilter(currentUser?.id, feedFilter, undefined, threadType)
+      .then((res: AxiosResponse) => {
+        setNotifications(res.data.data);
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          jsonData['api-error-messages']['fetch-activity-feed-error']
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx:89]: Can we use notification specific message here?

File: openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.utils.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+import Icon from '@ant-design/icons';
+import React from 'react';
+import { ReactComponent as IconTask } from '../../assets/svg/ic-task.svg';
+
+export const tabsInfo = [
+  {
+    name: 'Task',
+    key: 'Task',
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.utils.tsx:21]: Can we create an enum for the tabs key and use it everywhere?

File: openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  );
+
+  useEffect(() => {
+    getNotificationData(ThreadType.Task, FeedFilter.ASSIGNED_TO);
+  }, []);
+
+  const getTabTitle = (name: string, key: string) => {
+    return (
+      <Badge
+        dot={key === 'Task' ? hasTaskNotification : hasMentionNotification}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx:136]: Can we take this value from the function param like how we taking name and key?
```suggestion
        dot={hasNotification}
```

File: openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationFeedCard.component.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+const NotificationFeedCard: FC<NotificationFeedProp> = ({
+  createdBy,
+  entityFQN,
+  entityType,
+  timestamp,
+  feedType,
+  taskDetails,
+}) => {
+  const entityDisplayName = () => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationFeedCard.component.tsx:39]: Can we move this to feedUtils?

File: openmetadata-ui/src/main/resources/ui/src/components/nav-bar/NavBar.tsx [hunk 6]
```diff
@@ -66,7 +88,112 @@ const NavBar = ({
...
+              setHasMentionNotification(false);
+            }, 5000);
+
+          break;
+      }
+    }
+  };
+
+  const handleTaskNotificationRead = () => {
+    setHasTaskNotification(false);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/nav-bar/NavBar.tsx:114]: We are doing the same thing in line number `97` and `105`, can we move these functions up and use the same there as well?

**Review by Sachin-chaurasiya — APPROVED** [review]
Thanks, @chirag-madlani.

**Pull request merged**